functional-test/src/main/java/org/zanata/workflow/LoginWorkFlow.java

```diff
@@ -79,14 +79,16 @@ private void doSignIn(String username, String password) {
             .waitForAMoment().until(new Predicate<WebDriver>() {
             @Override
             public boolean apply(WebDriver driver) {
-                List<WebElement> messages =
-                        driver.findElements(By.className("message--danger"));
-                if (messages.size() > 0 && messages.get(0)
-                        .getText().contains(" Login failed ")) {
-                    throw new IllegalAccessError("Login failed");
-                }
-                List<WebElement> signIn = driver.findElements(By.id("Sign_in"));
-                return signIn.size() == 0;
+                // only enable this if you temporarily disable implicit waits:
```

Comment:
Actually, I'm no longer certain you can change the implicit wait.  I think
you use it all the time, or not at all.

On 26 February 2015 at 14:02, Damian Jansen <notifications@github.com>
wrote:

> In functional-test/src/main/java/org/zanata/workflow/LoginWorkFlow.java
> <https://github.com/zanata/zanata-server/pull/685#discussion_r25403485>:
>
> > @@ -79,14 +79,16 @@ private void doSignIn(String username, String password) {
> >              .waitForAMoment().until(new Predicate<WebDriver>() {
> >              @Override
> >              public boolean apply(WebDriver driver) {
> > -                List<WebElement> messages =
> > -                        driver.findElements(By.className("message--danger"));
> > -                if (messages.size() > 0 && messages.get(0)
> > -                        .getText().contains(" Login failed ")) {
> > -                    throw new IllegalAccessError("Login failed");
> > -                }
> > -                List<WebElement> signIn = driver.findElements(By.id("Sign_in"));
> > -                return signIn.size() == 0;
> > +                // only enable this if you temporarily disable implicit waits:
>
> It's unfortunate there's no way to query the driver's implicitWait value...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/685/files#r25403485>.
>